include/SH3/graphics/msbmp.hpp

```diff
@@ -0,0 +1,114 @@
+ *  @note All paths specified by the constructor are PHYSICAL paths on disk.
+ *  @note The palette data is stored as BGR with the last byte being 0xFF (unused)
+ *
+ *  @author Jesse Buhagiar
+ */
+#ifndef MSBMP_HPP_INCLUDED
+#define MSBMP_HPP_INCLUDED
+
+#include <cstdint>
+#include <vector>
+#include <string>
+
+#include <GL/gl.h>
+
+namespace sh3_graphics
+{
+    /** @defgroup graphics-headers Graphics headers
+     *  @{
+     */
+
+     #pragma pack(push, 1) // Stop the compiler from giving us stupid naturally aligned boundaries (__attribute__((packed)) didn't work...)
+
+    /**
+     * Bitmap file information header.
+     */
+    typedef struct
+    {
+        char tag[2];                /**< This is always the ASCII characters "BM" */
+        std::uint32_t fsize;        /**< The size of this file in bytes */
+        std::uint16_t rsvd1;        /**< 2 reserved bytes. Determined by program that created this bmp */
+        std::uint16_t rsvd2;        /**< 2 reserved bytes. Determined by program that created this bmp */
+        std::uint32_t pix_offset;   /**< Offset to beginning of image data */
+    } bmp_file_header;
+
+    /**
+     * Bitmap information header. Contains information about the actual bitmap (such as width and height)
+     */
+    typedef struct
+    {
+        std::uint32_t header_size;  /**< Size of the header following this one. Used to determine which one to use. */
+        std::int32_t width;         /**< Width of this image. For some reason it's a SIGNED 32-bit integer.... */
+        std::int32_t height;        /**< Height of this image. */
+        std::uint16_t colour_plane; /**< Number of colour planes. Should be 1. */
+        std::uint16_t bpp;          /**< Number of bits per pixel (for us, should be 8 or 24) */
+        std::uint32_t compression;  /**< Compression method used. THIS SHOULD ALWAYS BE 0 FOR UNCOMPRESSED!!!! */
+        std::uint32_t image_size;   /**< Size of this image in bytes */
+        std::int32_t x_res;         /**< Horizontal resolution in pixels per metre */
+        std::int32_t y_res;         /**< Vertical resolution in pixels per metre */
+        std::uint32_t palette_size; /**< Number of colours in the colour palette */
+        std::uint32_t imp_colours;  /**< Number of important colours used. 0 when every colour is important. Usually ignored */
+    } bmp_info_header;
+
+    #pragma pack(pop)
+
+    /**@}*/
+
+    /**
+     *  Describes a Microsoft bitmap file, which can be loaded and sent to the graphics card.
+     *  Note that all bmp files are stored on disk PHYSICALLY, therefore do not require the use
+     *  of the virtual filesystem.
+     */
+    struct msbmp final
+    {
+    public:
+        msbmp(const std::string& path){Load(path);}
+        ~msbmp(){};
+
+        /**
+         *  Load all image data
+         *
+         *  @param path Physical path to the file (bmps are NOT contained in an arc section)
+         */
+        int Load(const std::string& path);
```

Comment:
The return parameter should be documented.
It's an error, for which I'd prefer an `error` (from `error.hpp`).